File: CoffeeShop_Web/CoffeeShop/src/main/java/com/example/CoffeeShop/entity/OrderItem.java
package com.example.CoffeeShop.entity;

import com.google.firebase.database.PropertyName;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@NoArgsConstructor
@AllArgsConstructor
public class OrderItem {
    private String itemId;
    private String itemName;
    private int quantity;
    private double price;
    private boolean served;
    private String timestamp;
    private boolean sentToKitchen;

    // ÉP FIREBASE ĐỌC ĐÚNG TRƯỜNG "item_id" CÓ DẤU GẠCH DƯỚI TỪ JSON
    @PropertyName("item_id")
    public String getItemId() {
        return itemId;
    }

    @PropertyName("item_id")
    public void setItemId(String itemId) {
        this.itemId = itemId;
    }

    // ÉP FIREBASE ĐỌC ĐÚNG TRƯỜNG "item_name" CÓ DẤU GẠCH DƯỚI TỪ JSON
    @PropertyName("item_name")
    public String getItemName() {
        return itemName;
    }

    @PropertyName("item_name")
    public void setItemName(String itemName) {
        this.itemName = itemName;
    }

    // ÉP FIREBASE ĐỌC ĐÚNG TRƯỜNG "sentToKitchen" (Hoặc sent_to_kitchen nếu đổi)
    @PropertyName("sentToKitchen")
    public boolean isSentToKitchen() {
        return sentToKitchen;
    }

    @PropertyName("sentToKitchen")
    public void setSentToKitchen(boolean sentToKitchen) {
        this.sentToKitchen = sentToKitchen;
    }
}

File: CoffeeShop_Moblie/app/src/main/java/com/example/coffeeshopmobile/data/model/OrderItem.kt
package com.example.coffeeshopmobile.data.model

import com.google.firebase.database.PropertyName

data class OrderItem(
    @get:PropertyName("item_id")
    @set:PropertyName("item_id")
    var itemId: String = "",

    @get:PropertyName("item_name")
    @set:PropertyName("item_name")
    var itemName: String = "",

    var quantity: Int = 1,
    var price: Double = 0.0,
    var served: Boolean = false,
    var timestamp: String = "",

    // THÊM BIẾN NÀY: Để biết món đã bấm nút gửi xuống quầy bar chưa
    var sentToKitchen: Boolean = false
) {
    val total: Double
        get() = price * quantity
}